pkg/docker/client.js

```diff
@@ -726,6 +729,18 @@
             return connected.promise();
         };
 
+        this.containers_for_image = function containers_for_image(id) {
+            util.docker_debug('containers search on image id: ', id);
+            return http.get('/v1.12/containers/json', { all: 1 , filters: JSON.stringify({ ancestor: [ id ] }) })
+                .fail(function(ex) {
+                    console.log(ex);
+                    util.docker_debug('containers search on image id failed:', id, ex);
+                })
+                .done(function(data) {
+                    util.docker_debug('containers search on image id succeeded:', id);
+                }).then(JSON.parse);
```

Comment:
Does this `.then` really only apply to the "done" case? This is a bit ambiguous. Might be safer and also easier to read to just call `JSON.parse(data)` within `done()` and return it?

Reply:
I saw it in other places in the file, but yea you're right that returning it from `done()` is a lot clearer.